### PR TITLE
fix circular dependency on authn/user

### PR DESCRIPTION
**Pull request by Schparky**
Storage code borrowed from silinternational/wecarry-ui

### CODE REVIEW

File: src/data/storage.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    const fn = customClearFns.pop()
+    if (typeof fn === 'function') {
+      fn()
+    }
+  }
+}
+
+const customClearFns: Array<clearFn> = []
+
+interface clearFn {
```
Comment (RussellChamp) [inline src/data/storage.ts:17]: by convention, interfaces (much like classes and types) should start with a capital letter

File: src/data/storage.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export function clearApp(): void {
+  while (customClearFns.length) {
+    const fn = customClearFns.pop()
+    if (typeof fn === 'function') {
+      fn()
+    }
+  }
+}
+
+const customClearFns: Array<clearFn> = []
```
Comment (RussellChamp) [inline src/data/storage.ts:15]: while both syntax are valid, we have standardized on using `T[]` for array typing

File: src/data/storage.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
+export function onClear(fn: clearFn): void {
```
Comment (RussellChamp) [inline src/data/storage.ts:2]: if the intent is that the `clearFn`s are only to be executed _once_ and then removed from the array (as it appears to do in `clearApp` it should be noted in an inline comment on one of these lines as the behavior could easily be misunderstood by "the next guy" who need to make a change to the code.

**Review by RussellChamp — CHANGES_REQUESTED** [review]
looks mostly good

**Review by RussellChamp — COMMENTED** [review]
lgtm

**Review by hobbitronics — APPROVED** [review]
Nice

**Pull request merged**